### PR TITLE
Improve error message when function not found in driver runtime

### LINKED ISSUE #1585: Terrible error when nimi-python module tries to call a function not supported by driver runtime
### Description of issue

If the system has an old driver runtime installed and a newer nimi-python module, and the client tries to use a method that was added later to the driver runtime, then the error is not helpful: You get an `AttributeError` with no indication of what the problem is.

We should improve the error so that clear guidance is given to the user for correcting it.

### PR DESCRIPTION
**Pull request by ni-erikcrank**
- [x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/nimi-python/blob/master/CONTRIBUTING.md).
- [ ] ~I've updated [CHANGELOG.md](https://github.com/ni/nimi-python/blob/master/CHANGELOG.md) if applicable.~
- [ ] ~I've added tests applicable for this pull request~

### What does this Pull Request accomplish?

When a required function (e.g. CommitWithChannels) is missing from the driver runtime, nimi-python adds an error description that is more descriptive than "function not found" to instruct the user to upgrade to a newer version of the driver.

### List issues fixed by this Pull Request below, if any.

- Fix #1585 

### What testing has been done?

Called abort and commit methods on a system with NI-DCPower 20.5 instrument driver and confirmed that the error message is displayed in the exception information at runtime.

### CODE REVIEW

File: build/templates/_library.py.mako [hunk 1]
```diff
@@ -51,7 +51,12 @@ class Library(object):
     def ${c_func_name}(${param_names_method}):  # noqa: N802
         with self._func_lock:
             if self.${c_func_name}_cfunc is None:
-                self.${c_func_name}_cfunc = self._library.${c_func_name}
+                try:
+                    self.${c_func_name}_cfunc = self._library.${c_func_name}
+                except AttributeError as e:
+                    raise AttributeError("A required function was not found in the instrument driver DLL. This might "
+                                         "be an indication that the version of the instrument driver is too old for "
+                                         "this version of the python API. Upgrade your instrument driver.") from e
```
Comment (ni-erikcrank) [inline build/templates/_library.py.mako:59]: Please provide the preferred message text.

**Comment by sbethur** [comment]
ok to test

**Comment by marcoskirsch** [comment]
We use the term "driver runtime" to refer to the driver API called by nimi-python, as opposed to synonyms like "instrument driver DLL". Please correct description and other instances to be consistent.

**Comment by marcoskirsch** [comment]
Please referrence #1585 in Description.

File: build/templates/_library.py.mako (hunk 1 above)
Comment (marcoskirsch) [reply build/templates/_library.py.mako:59]: I think you have enough context in the codegenerator to give a really specific error. Here's a suggestion of what that should look like based on the error string in `DriverNotInstalledError`:

> Function `niDCPower_InitializeWithChannels` was not found in the NI-DCPower runtime. Please visit http://www.ni.com/downloads/drivers/ to download a newer version and install it.' 

An additional question: do we want a `DriverTooOld` class for this or is `AttributeError` better?

**Review by marcoskirsch — CHANGES_REQUESTED** [review]

**Comment by marcoskirsch** [comment]
@sbethur @ni-erikcrank we should be able to unit test in `nifake` by mocking ctypes, no?

File: build/templates/_library.py.mako (hunk 1 above)
Comment (ni-erikcrank) [reply build/templates/_library.py.mako:59]: I don't have an opinion on the DriverTooOld class.
Do you know of an existing way to obtain the driver name (e.g. NI-DCPower) for this message?

**Comment by sbethur** [comment]
test this please

File: build/templates/_library.py.mako (hunk 1 above)
Comment (ni-erikcrank) [reply build/templates/_library.py.mako]: I added a DriverTooOldError class.

**Comment by ni-erikcrank** [comment]
Added a unit test for the new LibraryTooOldError.

**Review by sbethur — APPROVED** [review]

**Review by marcoskirsch — APPROVED** [review]

**Pull request merged**